CONTRIBUTING.rst

```diff
@@ -19,8 +22,48 @@ Contributor Guide
 
    * Now, you have remote repositories named:
 
-      - ``upstream``, which refers to the ``networkx`` repository
-      - ``origin``, which refers to your personal fork
+     - ``upstream``, which refers to the ``networkx`` repository
+     - ``origin``, which refers to your personal fork
+
+   * Next, you need to set up your build environment.
+     Here are instructions for two popular environment managers:
+   
+     * ``venv`` (pip based)
+     
+       ::
+     
+         # Create a virtualenv named ``networkx-dev`` that lives in the directory of
+         # the same name
+         python -m venv networkx-dev
+         # Activate it
+         source networkx-dev/bin/activate
+         # Install all development and runtime dependencies of networkx
+         pip install -r <(cat requirements/*.txt)
```

Comment:
Perhaps it's worth adding a note about some of the requirements in `extras.txt` having external dependencies, e.g. graphviz and gdal, and that the installation may fail if these libraries are not available.

Reply:
I am hoping to fix both those issues before the 2.5 release, but I'll add a note for now.